lib/assert.js

```diff
@@ -122,7 +122,11 @@ function wrapAssertions(callbacks) {
 			} else if (isObservable(fn)) {
 				promise = observableToPromise(fn);
 			} else if (typeof fn !== 'function') {
-				throw new TypeError('t.throws must be called with a function, Promise, or Observable');
+				fail(this, new AssertionError({
+					actual: fn,
+					message: '`t.throws()` must be called with a function, Promise, or Observable'
+				}));
```

Comment:
I just tried and you're right. It is the existing behavior. I still think user mistakes like this should fail the test right away no matter what flags the user used though, but that's a separate discussion.